### ISSUE TITLE
error in running merge_bn.py

### ISSUE THREAD

**Issue by meixiansen** [chuanqi305/MobileNet-SSD#131]
when I ran python merge_bn.py --model example/MobileNetSSD_train.prototxt --weights snapshot/mobilenet_iter_40000.caffemodel 

got error:
F1012 13:44:31.570709 20815 insert_splits.cpp:29] Unknown bottom blob 'data' (layer 'conv0', bottom index 0)

**Comment by yesyu** [comment]
> when I ran python merge_bn.py --model example/MobileNetSSD_train.prototxt --weights snapshot/mobilenet_iter_40000.caffemodel
> 
> got error:
> F1012 13:44:31.570709 20815 insert_splits.cpp:29] Unknown bottom blob 'data' (layer 'conv0', bottom index 0)

same error, have you solved it?

**Comment by meixiansen** [comment]
> > when I ran python merge_bn.py --model example/MobileNetSSD_train.prototxt --weights snapshot/mobilenet_iter_40000.caffemodel
> > got error:
> > F1012 13:44:31.570709 20815 insert_splits.cpp:29] Unknown bottom blob 'data' (layer 'conv0', bottom index 0)
> 
> same error, have you solved it?
I found the previous version of merge_bn.py， then no error

**Comment by chuanqi305** [comment]
@meixiansen that's my fault, I forgot to change the example/MobileNetSSD_deploy.prototxt. And in new version you should use  python merge_bn.py --model example/MobileNetSSD_deploy.prototxt --weights snapshot/mobilenet_iter_40000.caffemodel to generate no_bn.prototxt and no_bn.caffemodel.

**Comment by wangpengabc** [comment]
I change the genetator for deploy.prototxt to make it generate the deploy model with BatchNorm layer and bias. Then run the merge_bn.py and generate no_bn.prototxt and no_bn.caffemodel successfully！

**Comment by WW2401** [comment]
Hi, the top name and bottom name of BatchNorm layer must be same? there is a BatchNorm layer whose top name is not same as bottom name in the pretrained model I used. T_T @chuanqi305
